### PR TITLE
Removed xfail label in scraper test & enabled to scrap local files

### PR DESCRIPTION
**Pull request by abhijeetmanhas**
`test_scraper.py` earlier had a test labelled `@pytest.mark.xfail`, so removed it. And added a new method `_localfilelist` to scrap local data files.
The method is called for `file://` protocols.

### CODE REVIEW

**Comment by abhijeetmanhas** [comment]
It is failing because file system in Windows is different from linux or Unix. Need to check whether sunpy rootdir need to be changed or some os specific changes in my function.

**Comment by abhijeetmanhas** [comment]
MacOS azure is stuck at installing tox.

File: changelog/3994.feature.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Added `_localfilelist` method in `~sunpy.util.scraper.Scraper` to scrap local data archives.
```
Comment (dstansby) [inline changelog/3994.feature.rst:1]: This isn't a user facing change, so not sure it needs to be a changelog. Is this fixing a bug? If so I would mention the bug that has been fixed here instead.

Comment (nabobalis) [reply changelog/3994.feature.rst:1]: It feels like a mix of the two.

**Comment by nabobalis** [comment]
Thanks again @abhijeetmanhas

**Pull request merged**